dipy/reconst/tests/test_qtdmri.py

```diff
@@ -0,0 +1,171 @@
+import numpy as np
+from dipy.data import get_gtab_taiwan_dsi
+from numpy.testing import (assert_almost_equal,
+                           assert_array_almost_equal,
+                           assert_equal,
+                           run_module_suite)
+from dipy.reconst.shore_time import ShoreTemporalModel
+from dipy.reconst import maptime
+from dipy.sims.voxel import (MultiTensor, all_tensor_evecs,  multi_tensor_pdf)
+from scipy.special import gamma
+from scipy.misc import factorial
+from dipy.data import get_sphere
+from dipy.sims.voxel import add_noise
+import scipy.integrate as integrate
+import scipy.special as special
+from dipy.core.gradients import gradient_table
+
+
+def generate_gtab4D(number_of_tau_shells=4):
+    gtab = get_gtab_taiwan_dsi()
+    qvals = np.tile(gtab.bvals / 100., number_of_tau_shells)
+    bvecs = np.tile(gtab.bvecs, (number_of_tau_shells, 1))
+    pulse_separation = []
+    for ps in np.linspace(0.02, 0.05, number_of_tau_shells):
+        pulse_separation = np.append(pulse_separation,
+                                     np.tile(ps, gtab.bvals.shape[0]))
+    pulse_duration = np.tile(0.01, qvals.shape[0])
+    gtab_4d = gradient_table(qvals=qvals, bvecs=bvecs,
```

Comment:
I created the gradient_table_from_qvals_bvecs function for this reason. I now use that new function to generate the 4D acquisition scheme here.